account/accounts.go

```diff
@@ -466,3 +466,21 @@ func (m *Manager) unlockExtendedKey(address, password string) (*SelectedExtKey,
 
 	return selectedExtendedKey, nil
 }
+
+// SignHash returns the signature for specified hash for the selected chat account
+func (m *Manager) SignHash(hash []byte) ([]byte, error) {
+	errorFormat := "SignHash failed: %s"
+
+	chatAccount, err := m.SelectedChatAccount()
+
+	if err != nil {
+		return []byte{}, fmt.Errorf(errorFormat, err.Error())
+	}
+
+	keyStore, err := m.geth.AccountKeyStore()
+	if err != nil {
+		return []byte{}, fmt.Errorf(errorFormat, "cannot get account keystore")
+	}
+
+	return keyStore.SignHash(chatAccount.GethAccount(), hash)
```

Comment:
I think the key is not found here which returns ErrLocked
```golang
// SignHash calculates a ECDSA signature for the given hash. The produced
// signature is in the [R || S || V] format where V is 0 or 1.
func (ks *KeyStore) SignHash(a accounts.Account, hash []byte) ([]byte, error) {
	// Look up the key to sign with and abort if it cannot be found
	ks.mu.RLock()
	defer ks.mu.RUnlock()

	unlockedKey, found := ks.unlocked[a.Address]
	if !found {
		return nil, ErrLocked
	}
	// Sign the hash using plain ECDSA operations
	return crypto.Sign(hash, unlockedKey.PrivateKey)
}
```

Reply:
ftr I am logged in in the app

Reply:
gotcha, I'll take a look today

Reply:
I would not use `geth.AccountKeyStore()` at all. `chatAccount, err := m.SelectedChatAccount()` should be enough to get the private key (`chatAccount.AccountKey.PrivateKey`) and we can call `crypto.Sign(hash, unlockedKey.PrivateKey)` directly.

Reply:
yeah, I did exactly that.